v21.2/crdb-internal.md

```diff
@@ -5,128 +5,642 @@ toc: true
+~~~
+
+### `index_usage_statistics`
+
+Contains one row for each index in the current database surfacing usage statistics for that specific index. This view is updated every time a transaction is committed. Each user-submitted statement on the specified index is counted as a use of that index and increments corresponding counters in this view. System and internal queries (such as scans for gathering statistics) are not counted.
+
+Column | Type | Description
+------------|-----|------------
+`table_id` | `INT8` | Unique table identifier.
+`index_id` | `INT8` | Unique index identifier.
+`total_reads` | `INT8` | Number of times an index was selected for a read.
+`last_read` | `TIMESTAMPTZ` | Time of last read.
+
+#### Examples
+
+##### Are there any indexes that have become stale and are no longer needed? Which indexes haven't been used during the past week?
+
+{% include_cached copy-clipboard.html %}
+~~~sql
+SELECT  ti.descriptor_name as table_name, ti.index_name, total_reads
+FROM  crdb_internal.index_usage_statistics AS us
+JOIN crdb_internal.table_indexes ti
+ON us.index_id = ti.index_id AND us.table_id = ti.descriptor_id
+WHERE last_read < NOW() - INTERVAL '1 WEEK'
+ORDER BY total_access desc;
 ~~~
 
+##### Which indexes are infrequently used? Are there any unused indexes?
+
+{% include_cached copy-clipboard.html %}
+~~~sql
+SELECT  ti.descriptor_name as table_name, ti.index_name, total_reads
+FROM crdb_internal.index_usage_statistics AS us
+JOIN crdb_internal.table_indexes ti
+ON us.index_id = ti.index_id AND us.table_id = ti.descriptor_id
+WHERE total_reads = 0;
+~~~
+
+### `statement_statistics`
+
+Column | Type | Description
+------------|-----|------------
+`aggregated_ts` | `TIMESTAMPTZ NOT NULL` | The time that statistics aggregation started.
+`fingerprint_id` | `BYTES NOT NULL` | Unique identifier of the statement statistics. This is constructed using the statement fingerprint text, and statement metadata (e.g. query type, database name, etc.)
+`transaction_fingerprint_id` | `BYTES NOT NULL` | Uniquely identifies a transaction statistics. The transaction fingerprint ID that this statement statistic belongs to.
+`plan_hash` | `BYTES NOT NULL` | Uniquely identifies a query plan that was executed by the current statement. The query plan can be retrieved from the `sampled_plan` column.
+`app_name` | `STRING NOT NULL`| The name of the application that executed the statement.
+`metadata` | `JSONB NOT NULL` | Metadata that describes the statement. See [`metadata` column](#metadata-column).
+`statistics` | `JSONB NOT NULL` | Statistics for the statement. See [`statistics` column](#statistics-column).
+`sampled_plan` | `JSONB NOT NULL` | The sampled query plan of the current statement statistics. This column is unfilled if there is no sampled query plan.
+`aggregation_interval` | `INTERVAL NOT NULL` | The interval over which statistics are aggregated.
+
+### `metadata` column
+
+Column | Type | Description
```

Comment:
nit: the `metadata` column is a JSONB column. A JSONB blob contains multiple *field*s . I think `Field` here would better than `Column`. (Since JSONB doesn't have the concept of Column)

Reply:
Fixed.